web/src/lib/domain-logic/shouldShowDisclaimerForProfileNotSubmittingData.test.ts

```diff
@@ -0,0 +1,102 @@
+import { shouldShowDisclaimerForProfileNotSubmittingData } from "@/lib/domain-logic/shouldShowDisclaimerForProfileNotSubmittingData";
+import { BusinessPersona, businessPersonas } from "@businessnjgovnavigator/shared/profileData";
+import {
+  generateBusiness,
+  generateFormationData,
+  generateProfileData,
+} from "@businessnjgovnavigator/shared/test";
+
+import { IsAuthenticated } from "@/lib/auth/AuthContext";
+
+const nonOwningPersonas: BusinessPersona[] = ["STARTING", "FOREIGN"];
+
+describe("shouldShowDisclaimerForProfileNotSubmittingData", () => {
+  it("returns true if business is not defined", () => {
+    expect(shouldShowDisclaimerForProfileNotSubmittingData(undefined, IsAuthenticated.TRUE)).toBe(true);
+    expect(shouldShowDisclaimerForProfileNotSubmittingData(undefined, IsAuthenticated.FALSE)).toBe(true);
+  });
+
+  it("returns true if isAuthenticated is UNKNOWN", () => {
+    const business = generateBusiness({
+      formationData: generateFormationData({
+        completedFilingPayment: false,
+      }),
+      profileData: generateProfileData({
+        dateOfFormation: undefined,
+        businessPersona: "STARTING",
+      }),
+    });
+    expect(shouldShowDisclaimerForProfileNotSubmittingData(business, IsAuthenticated.UNKNOWN)).toBe(true);
+    expect(shouldShowDisclaimerForProfileNotSubmittingData(business, IsAuthenticated.UNKNOWN)).toBe(true);
+  });
+
+  it("returns true if persona is OWNING and not Authenticated", () => {
```

Comment:
Boulder: see comment in Eng Channel, we can chat more about this with the broader team on Monday. It would be great if you can follow the `red, green, refactor` approach to see what tests need to be staged correctly.

Reply:
Yeah, in my first iteration I thought we would test via operating phase to know whether a user had logged in (GUEST mode operating phases), guess I missed this test when refactoring to Authentication. My bad